tests/integration/test_workflow_db_mgr.py

```diff
@@ -127,3 +128,53 @@ async def test_workflow_param_rapid_toggle(
 
         w_params = dict(schd.workflow_db_mgr.pri_dao.select_workflow_params())
         assert w_params['is_paused'] == '0'
+
+
+async def test_record_only_non_clock_triggers(flow, run, scheduler):
+    """Database does not record wall_clock xtriggers.
+
+    https://github.com/cylc/cylc-flow/issues/5911
+
+    Includes:
+        - Not in DB: A normal wall clock xtrigger (wall_clock).
+        - In DB: An xrandom mis-labelled as wall_clock trigger  DB).
+        - Not in DB: An execution retry xtrigger.
+
+    @TODO: Refactor to use simulation mode to speedup after Simulation
+    mode upgrade bugfixes: This should speed this test up considerably.
+    """
+    id_ = flow({
+        "scheduler": {
+            'cycle point format': '%Y'
+        },
+        "scheduling": {
+            "initial cycle point": "1348",
+            "xtriggers": {
+                "another": "xrandom(100)",
+                "wall_clock": "xrandom(100, _=Not a real wall clock trigger)",
+                "real_wall_clock": "wall_clock()"
+            },
+            "graph": {
+                "R1": "@another & @wall_clock & @real_wall_clock => foo"
+            }
+        },
+        'runtime': {
+            'foo': {
+                'execution retry delays': 'PT0S',
+                'script': (
+                    'test $CYLC_TASK_SUBMIT_NUMBER == 1 && exit 1 || exit 0')
+            }
+        }
+    })
+    # Run workflow unto completion:
+    schd = scheduler(id_, paused_start=False, run_mode='live')
+    async with run(schd) as log:
+        while 'Workflow shutting down - AUTOMATIC' not in log.messages:
+            await asyncio.sleep(0.5)
+
+    # Get xtriggers db table:
+    info = schd.workflow_db_mgr.get_pri_dao().conn.execute(
+        'SELECT * FROM xtriggers').fetchall()
+
+    # All xtriggers are xrandom: None are wall_clock:
+    assert all(i[0].startswith('xrandom') for i in info)
```

Comment:
Can simply use `db_select` fixture
```suggestion
    assert db_select(schd, False, 'xtriggers', 'signature') == [
        ('xrandom(100)',),
        ('xrandom(100, _=Not a real wall clock trigger)',)
    ]
```